### ISSUE TITLE
[DataTableSkeleton] Add prop to hide header in table skeleton

### ISSUE THREAD

**Issue by robherley** [carbon-design-system/carbon#5993]
### Summary

After #5752 was merged, the DataTableSkeleton component always has a header now, which is a change from how it used to look previously (which is understandable, given the v10 bump). 

For the React component, it would be nice to have a prop such as `showHeader` that can show or hide the extra divs, instead of having to apply some sort of CSS to hide them.

**Comment by asudoh** [comment]
Sounds close to #5967 - CC @carbon-design-system/design

**Comment by robherley** [comment]
Just to clarify, it's not the header row, it's the new header for the table (container with title & description)

**Comment by nictownsend** [comment]
This is a breaking change that was not communicated in the release changelog... I'm now forced to override with CSS behaviour that has been forced rather than added as an optional.

**Comment by nictownsend** [comment]
And it's definitely not a backwards compatible change, so by semver should have been a major version bump.

**Comment by nictownsend** [comment]
I'm happy to look at raising a PR if it's acceptable for it to be customisable

**Comment by asudoh** [comment]
@nictownsend My apologies for the change in table skeleton yielded a minor breaking change; The change was meant to be part of `v10` but the team slipped it through the crack, and it was seen not worth waiting for `v11`. A possible future consideration for this area is better release noting.

Wrt providing an option to hide title/description portion, technically it makes sense, but let's see what @carbon-design-system/design thinks from the point of whether our design guide since `v10` allows it.

**Comment by nictownsend** [comment]
Thank you - for now, the DataTable + DataTable skeleton are not consistent; I use a DataTable without a header or toolbar, and I will have to use CSS to hide the Skeleton header and toolbar for consistency.

If design do not believe the guide for `v10` allows a DataTable Skeleton without header/toolbar, then does that also mean the DataTable component (https://www.carbondesignsystem.com/components/data-table/usage) needs updating in the guide to state a Toolbar must always exist?

**Comment by richardpilot** [comment]
There are also use cases where you want the DataTable to be a skeleton but you have actions available in the toolbar that do not need a skeleton. For example, the primary button of a table being create new object, which can still occur whilst the underlying data is being refresh (causing a skeleton) e.g.

![image](https://user-images.githubusercontent.com/13081745/81457376-b6b5b780-918d-11ea-9b4b-baf744f17df4.png)

**Comment by GraceeFlower** [comment]
> There are also use cases where you want the DataTable to be a skeleton but you have actions available in the toolbar that do not need a skeleton. For example, the primary button of a table being create new object, which can still occur whilst the underlying data is being refresh (causing a skeleton) e.g.
> 
> ![image](https://user-images.githubusercontent.com/13081745/81457376-b6b5b780-918d-11ea-9b4b-baf744f17df4.png)

Hey team, is this question be resolved? I also wanna keep the search bar shown in skeleton status, any guidance? Thanks!
